---
title: "Initial Biome Composition Analysis"
author: "Timothy Brennhofer"
date: "June 12, 2018"
output: 
  html_document:
    code_folding: hide
---
#Differential Effects of Antibiotic Disturbance on Mouse Gut Microbiomes
***
### Initial Abundance Data Setup and Manipulation
***
##### We began by loading the mouse gut microbe abundance data provided by Aspen Reese and proceed to do the following:

1. Correct cell *130x4719* to show there was no recorded abundance
2. Filter out all samples where the total abundance was less than 5,000
3. Rarefy the data to account for differing sample sizes
4. Eliminated unnecessary metadata for ananalysis of initial community composition
5. Convert bacterial abundance to relative bacterial abundance within each host mouse
6. Define a new metadata variable for the samples titled __mouseid__ composed of the mouse's id number, whether or not it recieved the antibiotic treatment, and at which measurement hour it was taken


```{r setup, include = TRUE, messages = FALSE}
#load necessary packages -- "library" or "require"
require(tidyverse)
require(data.table)
require(vegan)

#load data
abun_raw <- fread("C:/Users/Timothy Brennhofer/Desktop/KBS REU/KBS Data/2016_10_13_run2_otus.csv")
abun <- as.data.frame(abun_raw)

#fix data entry problems
abun[130,4719] <- 0
abun$Cohoused[abun$Mouse == 3 & abun$Hour == 168] <- 'N'
abun$Cohoused[abun$Mouse == 6 & abun$Hour == 168] <- 'N'
abun$Cohoused[abun$Mouse == 7 & abun$Hour == 168] <- 'Y'
abun$Pair[abun$Mouse == 7 & abun$Hour == 168] <- 4
abun$Abx[abun$Mouse == 7 & abun$Hour == 168] <- 'Y'
abun$Dose[abun$Mouse == 7 & abun$Hour == 168] <- 'Recovery'
abun$Cohoused[abun$Mouse == 9 & abun$Hour == 168] <- 'Y'
abun$Pair[abun$Mouse == 9 & abun$Hour == 168] <- 2
abun$Abx[abun$Mouse == 9 & abun$Hour == 168] <- 'N'
abun$Dose[abun$Mouse == 9 & abun$Hour == 168] <- 'No'
abun$Cohoused[abun$Mouse == 12 & abun$Hour == 168] <- 'N'
abun$Cohoused[abun$Mouse == 16 & abun$Hour == 168] <- 'N'

#filter for only abundances over 5000 reads
filt <- rowSums(abun[, c(11:ncol(abun))]) >= 5000
abun <- abun[filt, ]

#rarefy
tmp <- rrarefy(abun[, c(11:ncol(abun))], 5000)
abun <- cbind(abun[, c(1:10)], tmp)
abun_bc_raw <- abun
# !!! Stop here to run BC Dissimilarity 

#check
all(rowSums(abun[, c(11:ncol(abun))]) == 5000)

#Select relevant metadata, create unique ID code for each mouse and measurement, make abundance relative
abun <- abun %>%
  select(-Run, -Day, -Reads, -`Cage#`, -Dose, -Cohoused, -Pair) %>%
  rename(Mouse = SampleID) %>%
  mutate(Mouse = as.factor(floor(Mouse))) %>%
  gather(otu, abun, - Abx, -Hour, -Mouse) %>%
  group_by(Mouse) %>%
  mutate(mouseid = paste(Mouse, Abx, Hour, sep = "_")) %>%
  mutate(abun = abun/sum(abun), otu = as.numeric(otu)) %>%
  ungroup()

```

### Adherence of Taxanomic Metadata to Abuandance Data
***

Here I take the taxonomical metadata previously compiled by John Guittar and join it with the previously organized abundance data via OTU identification numbers.

```{r adding taxonomy}

#Load taxonomical data
tax <- fread("C:/Users/Timothy Brennhofer/Desktop/KBS REU/KBS Data/otus_table_joined_taxa.txt")

#Organize tax data into traditional classifications
tax <- tax %>%
  rename(otu = `#OTU ID`, tax = taxonomy) %>%
  mutate(tax = gsub("\\s|.__|\\[|\\]|Other", "", tax)) %>%
  separate(tax, sep = ';', c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), fill = 'right')

#Convert unidentified taxonomical levels to "unclassified"
tax <- as.data.frame(apply(tax, 2, function(x) ifelse(x == '', 'unclassified', x)), stringsAsFactors = FALSE)
tax$otu <- as.numeric(tax$otu)

#Join taxonomical data and abundance data
abun_tax <- left_join(abun, tax, by = 'otu')

```

### Visualizing Community Compositions over the 120 Hr. Antibiotic Treatment Period
***

Using the abundance and taxonomic data, we can determine what bacteria compose the community and how that changes over the course of treatment.

##### This barchart does the following:
1. Divides the community at the phylum level
2. Includes only those phyla corresponding to at least 0.001% of the community
3.Segements the metacommunity by:
  + Individual mouse
  + Measurement hour (0, 25, 97, 120)
  + Presence of antibiotic treatment


```{r intital composition data}
#Construct stacked bar chart
#Filter for reads over 0.001% at desired time points
#compare mice recieving and not recieving antibiotics
initial_abun_tax <- abun_tax
initial_abun_tax %>%
  filter(abun > 0.00001 & Hour %in% c(0,25,97,120)) %>%
  ggplot(aes(x = Mouse, y = abun, fill = Phylum)) +
    geom_bar(stat = 'identity') +
    facet_grid(Hour~Abx, scales = "free")

```

### Homogeneity across Communities Decreases with Increasing Taxonomical Resolution
***

This required the manipulation of the taxonomic data, and that it be joined to the abundance data after these changes. Each taxonomic identifaction level needed to include all higher levels of identification in order to guaruntee "_unkown_" samples would be accounted for individually. 

##### __Remember:__ OTU-Level Issues

```{r decrease in homogeneity by taxonomical unit}

#Combine taxonomic level IDs so each ID is unique
#Join to the abundance data once again
j <- tax %>%
  ungroup() %>%
  transmute(
    Phylum = as.numeric(as.factor(Phylum)),
    Class = paste0(Phylum, as.numeric(as.factor(Class))),
    Order = paste0(Phylum, Class, as.numeric(as.factor(Order))),
    Family = paste0(Phylum, Class, Order, as.numeric(as.factor(Order))),
    Genus = paste0(Phylum, Class, Order, Family, as.numeric(as.factor(Genus))),
    Species = paste0(Phylum, Class, Order, Family, Genus, as.numeric(as.factor(Species))),
    OTU = paste0(Phylum, Class, Order, Family, Genus, Species, otu),
    otu = otu
    ) %>%
  gather(tax_level, name, -otu) %>% 
  full_join(abun, by = 'otu') %>%
  filter(abun > 0 & Hour <= 120) 

# Determine the total richness of all the communities and the richness of each host for each taxonomic level

j <- j %>% 
  mutate(tax_level = factor(tax_level, levels =  c('Phylum','Class','Order','Family','Genus','Species','OTU'))) %>%
  group_by(tax_level) %>% 
  mutate(total_rich = n_distinct(name)) %>% 
  group_by(tax_level, Mouse, Hour, Abx) %>% 
  summarise(rich = n_distinct(name), total_rich = unique(total_rich), H = (-((rich/total_rich)*log(rich / total_rich))))

#Here, (H)= Shannon Diversity Index          
#Alternatively, H = (rich/total_rich)) can be used to determine the simple proportion of present units among the total possible across the mice

#Graph using the simple presence proportion
j %>% 
  ungroup() %>%
  ggplot(aes(x = tax_level, y = (rich/total_rich), group = Mouse, color = Abx)) + 
  scale_color_manual(values = c("black", "red")) +
  geom_smooth(se = FALSE) + 
  expand_limits(y = c(0,1))

#Graph how Shannon Diversity Index flucuates over the treatment period
j %>%
  ungroup() %>%
  filter(tax_level == "OTU") %>%
  ggplot(aes(x = Hour, y = H, group = Mouse, color = Abx )) +
  scale_color_manual(values = c("black", "red")) +
  geom_smooth(se=FALSE) +
  expand_limits(y = c(0,0.5))
```

### BC Dissimilarity Changes over Time
***

Mice were re-labeled to include all relevant metadata.
Bray-Curtis Dissimilarity was calculated among all mice and within both the antibiotic treatments and treatment time points.


```{r dissimilarity analysis}

#Create mouseid variable to consolidate metadata as the row names
abun_bc <- abun_bc_raw %>%
  select(-Run, -Day, -Reads, -`Cage#`, -Dose, -Cohoused, -Pair) %>%
  rename(Mouse = SampleID) %>%
  mutate(Mouse = as.factor(floor(Mouse))) %>%
  mutate(mouseid = paste(Mouse, Abx, Hour, sep = "_")) %>%
  select(mouseid, everything()) %>%
  column_to_rownames(var = "mouseid")

#Build Bray-Curtis Matrix
dissim_abun_BC <- vegdist(abun_bc[c(4:ncol(abun_bc))], method = "bray")
BC_matrix <- as.matrix(dissim_abun_BC)
BC_matrix[upper.tri(BC_matrix, diag = TRUE)] <- NA
melted_bc_matrix <- melt(BC_matrix) %>%
  rename(BC = value)

#Separate meta-data into individual columns
melted_bc_matrix <- separate(melted_bc_matrix, Var1, sep = "_", c("Mouse1", "Abx1", "Hour1"))
melted_bc_matrix <- separate(melted_bc_matrix, Var2, sep = "_", c("Mouse2", "Abx2", "Hour2"))

#Filter for similar antibiotic treatments and similar time points during treatment only
melted_bc_matrix <- melted_bc_matrix %>%
  filter(Abx1 == Abx2 & Hour1 == Hour2 & !is.na(BC)) %>%
  ungroup() %>%
  mutate(Hour1 = as.numeric(Hour1), Hour2 = as.numeric(Hour2)) %>%
  subset(select = -c(Mouse2, Hour2, Abx2)) %>%
  rename(Mouse = Mouse1, Abx = Abx1, Hour = Hour1) %>%
  filter(Hour <= 120) %>%
  group_by( Abx, Hour)

#Graph results  
ggplot(melted_bc_matrix, aes(x = Hour, y = BC, group = Abx, color = Abx)) +
  scale_color_manual(values = c("black", "red")) +
  geom_point() +
  geom_smooth()

```

### Relative Performance 
***

Type info here


``` {r Relative Performance Analysis}
#filter instances only 0, fewer than 5 data points
rp_abun_tax_raw <- abun_tax %>%
  mutate(Entero_Sort = if_else(Family == "Enterobacteriaceae", "Y", "N")) %>%
  group_by(Abx, Mouse, Entero_Sort, otu) %>%
  filter(Hour <= 120 & sum(abun > 0) > 5) %>%
  do(mod = summary(lm(abun ~ Hour, .))) %>%
  mutate(
    pval = mod$coefficients[[8]],
    tval = mod$coefficients[[6]]) %>%
  filter(pval <= 0.05) %>%
  ungroup() %>%
  group_by(Abx, otu, Entero_Sort) %>%
  mutate(
    pos_tval = sum(tval > 0), 
    neg_tval = sum(tval < 0)) %>%
  filter(pos_tval + neg_tval > 1)

rp_abun_tax <- rp_abun_tax_raw %>%
  group_by(Abx, otu, Entero_Sort) %>%
  summarise(
    pos_tval = sum(tval > 0), 
    neg_tval = sum(tval < 0)) %>%
  filter(pos_tval + neg_tval > 1)
  

ggplot(rp_abun_tax, aes(x = pos_tval, y = neg_tval, color = Abx, shape = Entero_Sort)) +
  geom_jitter(width = .25, height = .25) +
  scale_color_manual(values = c("black", "red")) +
  scale_shape_manual(values = c(16, 8)) +
  geom_abline (intercept = 0, slope = 1, linetype = 2, color = "blue") +
  expand_limits(x = c(0,7.5), y = c(0,7.5))
  
#rp_abun_tax_filt <- rp_abun_tax_raw %>%
#IDing individual otus and models of interest
#Cluster 1 on Y-Axis
#filter(pos_tval == 0 & Abx == "N")
#Cluster 2 on X-Axis
#filter(neg_tval == 0 & Abx == "Y" & Entero_Sort == "N")
#Cluster 3 on Y-Axis
#filter(pos_tval == 0 & Abx == "N" & otu != 273657)

```
